src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1287,23 +1295,26 @@ private void EngineController_TraceReconcliationComplete(TraceReconciliationEven
 
                 // TODO: MAGN-7314
                 // Find the owning workspace for a node.
-                var nodeSpace =
-                    Workspaces.FirstOrDefault(
-                        ws =>
-                            ws.Nodes.FirstOrDefault(n => n.GUID == nodeGuid)
-                                != null);
+                if (!nodeToWorkspaceMap.TryGetValue(nodeGuid, out var nodeSpace))
```

Comment:
hmm... on first glance memoizing this lookup makes sense, but I'm unclear exactly how many times we recreate the map vs how fast we bail on first finding the node in the existing code -can you look into that?

Reply:
discussed on slack- seems the cost to create this map each run is likely worth it so the lookup is much faster later. I do still have question (asked above):

Do we need to put nodes from custom node workspaces into the map - I kind of remember that nodes in custom node workspaces do participate in trace, it's just that custom node workspaces don't serialize trace - maybe that does not matter here?

Can you look into this?